File: app/src/main/java/net/wojteksz128/worktimemeasureapp/ComeEventAdapter.java
package net.wojteksz128.worktimemeasureapp;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import net.wojteksz128.worktimemeasureapp.database.ComeEvent;

import java.util.List;

public class ComeEventAdapter extends RecyclerView.Adapter<ComeEventViewHolder> {

    private List<ComeEvent> events;


    @NonNull
    @Override
    public ComeEventViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);

        View view = inflater.inflate(R.layout.day_list_item, parent, false);
        return new ComeEventViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ComeEventViewHolder holder, int position) {
        holder.bind(events.get(position));
    }

    @Override
    public int getItemCount() {
        return events != null ? events.size() : 0;
    }

    public void setEvents(List<ComeEvent> events) {
        this.events = events;
        notifyDataSetChanged();
    }
}

File: app/src/main/kotlin/net/wojteksz128/worktimemeasureapp/window/main/ComeEventAdapter.kt
package net.wojteksz128.worktimemeasureapp.window.main

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import net.wojteksz128.worktimemeasureapp.R
import net.wojteksz128.worktimemeasureapp.database.comeEvent.ComeEvent

class ComeEventAdapter : RecyclerView.Adapter<ComeEventViewHolder>() {

    private var events: List<ComeEvent>? = null


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ComeEventViewHolder {

        val context = parent.context
        val inflater = LayoutInflater.from(context)

        val view = inflater.inflate(R.layout.main_day_event_list_item, parent, false)
        return ComeEventViewHolder(view)
    }

    override fun onBindViewHolder(holder: ComeEventViewHolder, position: Int) {
        holder.bind(events!![position])
    }

    override fun getItemCount(): Int {
        return if (events != null) events!!.size else 0
    }

    fun setEvents(events: List<ComeEvent>) {
        this.events = events
        notifyDataSetChanged()
    }
}
